### ISSUE TITLE
Can this pytorch version reproduce the performance of reported results?

### ISSUE THREAD

**Issue by densechen** [HuguesTHOMAS/KPConv-PyTorch#70]
Hi, @HuguesTHOMAS,
Thanks for providing such nice code. 

I trained the model in ModelNet40 task with the default arguments, but only got a ~91.5 % testing accuracy.
Are there any tips should I notice? 

By the way, how about the performance of other tasks?
See also:
 https://github.com/HuguesTHOMAS/KPConv-PyTorch/issues/38#issuecomment-758470984

**Comment by HuguesTHOMAS** [comment]
Hi @densechen,

First, there is a strange phenomenon happening with this dataset, the validation reaches a higher score than the testing (although they are performed on the same set). I have never found the explanation for this and since then I moved on to other projects. 

Second, you have to understand that ModelNet40 is not a perfect dataset, and in my opinion, the score does not reflect the network performances. I consider that when reaching more than 91%, a network has proven effective and you can move on to larger and more complex datasets. If you want a better score, you can scrap for the last 1 or 2% by optimizing parameters, but I consider it overfitting. Such highly optimized parameters for ModelNet40 would not give you better scores on other datasets.

Now regarding your question about the performances of this specific PyTorch implementation. Given that I changed some internal stuff from the original Tensorflow implementation, and did not take the time to optimize everything again, in the current state, this implementation does reach the same scores as the Tensorflow one but should be close. It has the same potential, it is just a question of time spent repeating experiments again and again and slightly optimizing the scores.

This repo is not aimed at people seeking to reproduce results. It is aimed at people that want to use KPConv for new stuff, modify it, improve it, and experiment with it.

Best,
Hugues

**Comment by densechen** [comment]
Hi, @HuguesTHOMAS ,
Thanks for your replay!

> First, there is a strange phenomenon happening with this dataset, the validation reaches a higher score than the testing (although they are performed on the same set). I have never found the explanation for this and since then I moved on to other projects.

I go through your code, but cannot find any script to specify the random seed. If the model is running in a  randomization process, such as the point sample process and others, the test accuracy will be always lower than validation one. You can regard that the training process is overfitting already.  According to your code, the voting process contains a randomization process, so, I thought this was the problem.



> Second, you have to understand that ModelNet40 is not a perfect dataset, and in my opinion, the score does not reflect the network performances. I consider that when reaching more than 91%, a network has proven effective and you can move on to larger and more complex datasets. If you want a better score, you can scrap for the last 1 or 2% by optimizing parameters, but I consider it overfitting. Such highly optimized parameters for ModelNet40 would not give you better scores on other datasets.

I agree with you that ModelNet40 may not be a perfect dataset, and many other methods, even do not release the code for it. But I cannot agree that 91% is good enough. The original version of PointNet can also achieve 91+% with same easy data argument tricks. As for many methods, it is hard to obtain 92+% actually, and we also found that sometimes the higher performance on ModelNet40 has no relation to the performance on other datasets. 

> Now regarding your question about the performances of this specific PyTorch implementation. Given that I changed some internal stuff from the original Tensorflow implementation, and did not take the time to optimize everything again, in the current state, this implementation does reach the same scores as the Tensorflow one but should be close. It has the same potential, it is just a question of time spent repeating experiments again and again and slightly optimizing the scores.
> 
> This repo is not aimed at people seeking to reproduce results. It is aimed at people that want to use KPConv for new stuff, modify it, improve it, and experiment with it.

Thanks for providing a so nice work again! Recently, I am working on this project and improving the performance of it. But, you know that, if this baseline could not achieve the same performance (or closely performance) of the paper's, it is not fair for us to compare with. Avoiding this, I retrained it on ModelNet40, (because it is the most easy dataset, and can be quickly validated). However, I got the unexpected result. It makes me worried that can this be a good baseline for us to do a further experiments. 

Best,
Chen
